drivers/net/wireless/marvell/mwifiex/main.c

```diff
@@ -226,6 +226,19 @@ static int mwifiex_process_rx(struct mwifiex_adapter *adapter)
 	return 0;
 }
 
+static void maybe_quirk_fw_disable_ds(struct mwifiex_adapter *adapter)
+{
+	struct mwifiex_private *priv = mwifiex_get_priv(adapter, MWIFIEX_BSS_ROLE_STA);
+	struct mwifiex_ver_ext ver_ext;
+
+	memset(&ver_ext, 0, sizeof(ver_ext));
+	ver_ext.version_str_sel = 1;
+	mwifiex_send_cmd(priv, HostCmd_CMD_VERSION_EXT,
+			 HostCmd_ACT_GEN_GET, 0, &ver_ext, false);
+
+	priv->ds_quirk_version_string_requested = true;
```

Comment:
Is it correct that you're setting this _after_ sending the command or could the command be theoretically completed in `mwifiex_ret_ver_ext()` before you set this to `true`?

Reply:
For clarity it might be a good idea to set `ds_quirk_version_string_requested` to `true` before sending the command, but to answer the question: No, we won't invoke into the command response handler synchronously, because `false` was passed as the `sync` argument of `mwifiex_send_cmd()`, which means the command gets pushed to the end of the command queue and will be executed later.

Reply:
But is there anything that'd prevent a scenario like:
- Code executes until right after `mwifiex_send_cmd()`.
- Kernel decides to un-schedule the current thread before `priv->ds_quirk_version_string_requested = true;` and schedules the (probably workqueue) thread processing the command that just got queued.
- Command gets executed and completed before the verext thread gets re-scheduled and the flag gets set to true.

Stuff like that is probably very unlikely due to the time it takes to talk to the firmware, but theoretically still possible, unless there's some other synchronization going on (like this code being executed on the same thread that processes the command workqueue).

Reply:
Oh yes, this code is being executed in the workqueue thread, I didn't mention that. That's actually exactly the reason why we can't run the command synchronously, because blocking while inside the loop that executes commands and gets responses is obviously not going to work.

Reply:
Got it, then that will work.